Samples/MessageZooEvents/ZooEventsConsumer.Tests/SavePetsTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿using PactNet.PactMessage;
+using Xunit;
+using ZooEventsConsumer.Models;
+using PactNet.PactMessage.Models;
+using PactNet.Matchers;
+using Xunit.Abstractions;
+
+namespace ZooEventsConsumer.Tests
+{
+    public class SavePetsTests : IClassFixture<ConsumerEventPact>
+    {
+        private readonly IMessagePact _messagePact;
+
+        public SavePetsTests(ConsumerEventPact data, ITestOutputHelper output)
+        {
+            _messagePact = data.Initialise(output);
+        }
+
+        [Fact]
+        public void Handle_WhenAPetIsCreated_SavesThePet()
+        {
+            //Arrange
+            var stubRepo = new StubPetRepo();
+            var consumer = new SavePets(stubRepo);
+
+            //TODO: Why is this an object?
+            var providerStates = new[]
+            {
+                new ProviderState
+                {
+                    Name = "there is a Pet animal",
+                }
+            };
```

Comment:
In the v3 message pact contract (v3 specification) the provider state was changed a little bit as stated here (https://github.com/pact-foundation/pact-specification/tree/version-3#allow-multiple-provider-states-with-parameters). Actually added a "params" property and deleted it because it wasn't supported. Let's add this property and update the known issue.

Reply:
Not sure actually, I think that I've tested it and it created multiple interactions with the same description and different provider state. Worth testing.